### PR TITLE
set up association

### PR DESCRIPTION
**Pull request by Richez003**
implemented the following :

Create a class Classroom with the following:
@label instance variable, should be initialized in the constructor.
Setter and getter for @label (remember about attr_accessor).
Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:
Create the has-many side (a classroom has many students).
Create the belongs-to side (a student belongs to a classroom).
Make sure that when adding a student to a classroom it also sets the classroom for the student.
Make sure that when setting the classroom for a student it also adds it to the classrooms' students.
Create a class Book with the following:
@title and @author instance variables, should be initialized in the constructor.
Setters and getters for instance variables (remember about attr_accessor).
Create a class Rental with the following:
@date instance variable, should be initialized in the constructor.
Setter and getter for @date (remember about attr_accessor).
Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:
Create the has-many side of Book and Rental (a book has many rentals).
Create the belongs-to side of Rental and Book (a rental belongs to a book).
Create the has-many side of Person and Rental (a person has many rentals).
Create the belongs-to side of Rental and Person (a rental belongs to a person).
Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
+  def initialize(label)
```
Comment (mikethreels) [inline classroom.rb]: - [x] there needs to be an attribute reader for student as well in this class.

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    @label = label
+    @student = []
+  end
+
+  def add_studnt(student)
+    @student.push?(student)
```
Comment (mikethreels) [inline classroom.rb]: - [x] please remove the '?' in this line as it will make your application not work.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  attr_accessor :author, :title, :rental
+
+  def initialize(author, title)
+    @author = author
+    @title = title
+    @rental = []
+  end
+
+  def add_rental(_person, date)
+    Rental.new(perason, date, self)
```
Comment (mikethreels) [inline book.rb]: - [x] it looks like you made a small typo in this line, it should be `person` instead of `perason`.

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class Rental
+  attr_accessor :date
```
Comment (mikethreels) [inline rental.rb:2]: - [x] in this class Getters and Setters for @book and @person instance variables should be present

File: rental.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+  def initiallize(date, book, _peson)
+    @date = date
+    @book = book
+    @person = person
+  end
```
Comment (mikethreels) [inline rental.rb]: - [ ] The constructor of Rental should be modified so Book and Person are set in it to break the many-to-many relationship. Link to the example article: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Richard, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Required Changes ♻️
- [x] in the Student class you should also add a method that serves as the 'belongs-to' side (a student belongs to a classroom). this is a link to the article with examples: https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classroom.rb (hunk 1 above)
Comment (Richez003) [reply classroom.rb]: done

File: classroom.rb (hunk 2 above)
Comment (Richez003) [reply classroom.rb]: done

File: book.rb (hunk 3 above)
Comment (Richez003) [reply book.rb]: done

File: rental.rb (hunk 5 above)
Comment (Richez003) [reply rental.rb]: done

File: rental.rb (hunk 4 above)
Comment (Richez003) [reply rental.rb:2]: done

**Review by arslanbisharat — APPROVED** [review]
Hi @Richez003,
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**